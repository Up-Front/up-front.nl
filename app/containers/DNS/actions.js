/* globals fetch */
export const SITES_INVALID = 'SITES_INVALID';
export const SITES_FETCHING = 'SITES_FETCHING';
export const SITES_FETCHED = 'SITES_FETCHED';
export const SITES_FETCH_FAILED = 'SITES_FETCH_FAILED';

function fetchDNS() {
  return (dispatch) => {
    dispatch({ type: SITES_FETCHING });

    return fetch('http://jsonplaceholder.typicode.com/users')
      .then(res => res.json())
      .then(result => dispatch({ type: SITES_FETCHED, result }))
      .catch(error => dispatch({ type: SITES_FETCH_FAILED, error }));
  };
}

function shouldfetchDNS({ sites }) {
  if (!sites.list ||
    sites.readyState === SITES_FETCH_FAILED ||
    sites.readyState === SITES_INVALID) {
    return true;
  }

  return false;
}

export function fetchDNSIfNeeded() {
  return (dispatch, getState) => {
    if (shouldfetchDNS(getState())) {
      return dispatch(fetchDNS());
    }
    return null;
  };
}
