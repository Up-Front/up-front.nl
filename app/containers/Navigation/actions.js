/* globals fetch */

import { doGet } from '../../api';

export const SITES_INVALID = 'SITES_INVALID';
export const SITES_FETCHING = 'SITES_FETCHING';
export const SITES_FETCHED = 'SITES_FETCHED';
export const SITES_FETCH_FAILED = 'SITES_FETCH_FAILED';

function fetchSites() {
  return (dispatch) => {
    dispatch({ type: SITES_FETCHING });

    return doGet('http://jsonplaceholder.typicode.com/users')
      .then(res => res.json())
      .then(result => dispatch({ type: SITES_FETCHED, result }))
      .catch(error => dispatch({ type: SITES_FETCH_FAILED, error }));
  };
}

function shouldfetchSites({ sites }) {
  if (!sites.list ||
    sites.readyState === SITES_FETCH_FAILED ||
    sites.readyState === SITES_INVALID) {
    return true;
  }

  return false;
}

export function fetchSitesIfNeeded() {
  return (dispatch, getState) => {
    if (shouldfetchSites(getState())) {
      return dispatch(fetchSites());
    }
    return null;
  };
}
