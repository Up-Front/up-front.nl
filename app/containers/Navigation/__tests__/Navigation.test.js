import  React from 'react'

const renderer = require('react-test-renderer');

import { Navigation } from '../index';
import * as SitesActions from '../actions';

const dispatch = (e) => {
    console.log('dispatched:', e);
}

const sites = {
    readyState: SitesActions.SITES_FETCHED,
    list: [
        {
            id: 0,
            name: 'Bakker Bollebof',
            url: 'bakkerbollebof.nl'
        },
        {
            id: 1,
            name: 'Sjakie Slager',
            url: 'sjakieslager.nl'
        },
    ]
}

it('renders correctly', () => {
  const tree = renderer.create(
    <Navigation sites={sites} dispatch={dispatch} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
