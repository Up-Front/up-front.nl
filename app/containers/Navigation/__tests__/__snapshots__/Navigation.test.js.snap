// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <ul>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        Bakker Bollebof
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        Sjakie Slager
      </a>
    </li>
  </ul>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    Home
  </a>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    DNS
  </a>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    Instellingen
  </a>
</div>
`;
